process/sync/shardForkDetector.go

```diff
@@ -69,51 +78,148 @@ func (sfd *shardForkDetector) AddHeader(
 		return err
 	}
 
-	sfd.activateForcedForkIfNeeded(header, state)
+	if header.GetNonce() > sfd.highestNonceReceived() {
```

Comment:
I would keep set to be set, if you call set with 5 than you will set 5 no matter what. The logic if you set or now should be outside

Reply:
it is setHighest, it can have verification inside. Whenever you call this verification is the same, so duplicated code. If you think you need forcedSet without verification, than add a parameter in the setHighestNonceReceived function like verifiy bool. If it is true than verification is done, otherwise set directly.

Reply:
Done